2024Robot_Team3484/src/main/cpp/subsystems/IntakeSubsystem.cpp

```diff
@@ -0,0 +1,92 @@
+//
+// Intake Subsystem
+//
+// -Noah, Aidan  & Ethan
+//
+
+#include <subsystems/IntakeSubsystem.h>
+#include <units/angle.h>
+
+using namespace IntakeConstants;
+
+IntakeSubsystem::IntakeSubsystem() {
+    _pivot_motor.RestoreFactoryDefaults();
+    _drive_motor.RestoreFactoryDefaults();
+
+    _drive_motor.SetIdleMode(rev::CANSparkMax::IdleMode::kBrake);
+
+    _pivot_pid_controller.SetFeedbackDevice(_pivot_encoder);
+
+    _target_position = STOW_POSITION;
+
+    _pivot_pid_controller.SetP(0.1);
+    _pivot_pid_controller.SetI(1e-4);
+    _pivot_pid_controller.SetD(1);
+    _pivot_pid_controller.SetIZone(0);
+    _pivot_pid_controller.SetFF(0);
+    _pivot_pid_controller.SetOutputRange(-1, 1);
+
+}
+
+void IntakeSubsystem::Periodic() {
+    // Runs every 20ms
+
+    const frc::TrapezoidProfile<units::degree>::State current_state{
+        GetIntakePosition(), 
+        units::revolutions_per_minute_t{_pivot_encoder.GetVelocity()}
+    }; // units::turns converts the revolutions to a angle value
+
+    const frc::TrapezoidProfile<units::degree>::State target_state{
+        _target_position,
+        0_deg_per_s
+    };
+
+    const units::turn_t linear_angle = _intake_trapezoid.Calculate(20_ms, current_state, target_state).position;
+
+    _pivot_pid_controller.SetReference(linear_angle.value(), rev::CANSparkMax::ControlType::kPosition);
+
+    if (ArmExtended() && !_arm_sensor_hit) {
```

Comment:
What is `arm_sensor_hit`? In the context of what I think this statement says, the home position sensor should never be true (thinking logically and ignoring actual normal state of sensor) when the arm is extended (unless something is wrong with the sensor) which means this statement will never run. I think what you want is:
```
if ( arm is at home )
{
    set encoder to 0 deg
    set PID setpoint to 0 deg
}
```

Alternatively, you can declare `linear_angle` as a non-const variable and set the the value of `linear_angle` to either `_intake_trapezoid.Calculate( )` _OR_ set it to 0 if the home position sensor is true, then call `_pivot_pid_controller.SetReference( )` at the end of the `Periodic( )` function.